schema/backwards_compatibility_test.go

```diff
@@ -100,8 +100,7 @@ func TestBackwardsCompatibilityImageIndex(t *testing.T) {
          "digest": "sha256:fb2fc0707b86dafa9959fe3d29e66af8787aee4d9a23581714be65db4265ad8a",
          "platform": {
             "architecture": "arm64",
-            "os": "linux",
-            "variant": "armv8"
```

Comment:
I have been chatting with the kernel/toolchain and disto teams here within ARM, and the consensus is that there is currently no reliable way of runtime detecting the underlying architecture.

To quote our kernel team when asked about runtime detection
"As an example, Linux reports ARMv7 for ARM11MPCore or ARM1176 (and I think newer revisions of ARM1136). The architecture versions are just some milestones or new baselines without a dedicated ID. The architected CPUID registers only expose features and not architecture versions. However, since some features are "back-ported" to previous
architecture versions, the kernel gets confused (the ARMv7 MMU features have been built into ARM11MPCore, hence Linux reporting v7).

The "armv8l" is _only_ reported by an AArch64 kernel when running a task with the PER_LINUX32 personality. Otherwise the utsname is "aarch64". If you run an AArch32 kernel on ARMv8, it will report armv7l.

I doubt the above helps but I think we should just avoid such run-time
detection at the architecture version. If the hardware does not provide
this information, the OS cannot reliably infer it."

I know this is not very helpful at the moment, but it is a statement of where we are today.  There is talk about ARM potentially investing time in 'lscpu' with the help of a LUT to correctly identify architecture.